rfcs/text/semantic-release.md

```diff
@@ -0,0 +1,257 @@
+> * don't capitalize the first letter
+> * no dot (.) at the end
+> ## Body
+> Just as in the *subject*, use the imperative, present tense: "change" not "changed" nor "changes". The body should include the motivation for the change and contrast this with previous behavior.
+>
+> ## Footer
+> The footer should contain any information about **Breaking Changes** and is also the place to reference GitHub issues that this commit **Closes**.
+>
+> **Breaking Changes** should start with the word `BREAKING CHANGE:` with a space or two newlines. The rest of the commit message is then used for this.  
+> [[Source]](https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit)
+
+We will use [`commitlint`](https://github.com/conventional-changelog/commitlint) and [`husky`](https://github.com/typicode/husky) to check our commits against the `conventionalcommits` guidelines.
+- [`husky`](https://github.com/typicode/husky) will manage the git commit hooks defined in `package.json`
+- we will write a hook that runs the commit message via [`commitlint`](https://github.com/conventional-changelog/commitlint)
+- we will use the [`@commitlint/config-conventional`](https://github.com/conventional-changelog/commitlint/tree/master/%40commitlint/config-conventional) config
+- [`husky`](https://github.com/typicode/husky) will abort the commit if the commit message is in the wrong format
+
+We will use [`commitizen`](https://github.com/commitizen/cz-cli) to help write commit messages
+- we will use the [`cz-conventional-changelog`](https://github.com/commitizen/cz-conventional-changelog) config
+- users can choose to write commit messages with the regular `git` CLI or in their IDE still
+- users can launch [`commitizen`](https://github.com/commitizen/cz-cli) with `npm run commit`
+
+We will add [`probot/semantic-pull-requests`](https://probot.github.io/apps/semantic-pull-requests/) as a GitHub check.
+This will prevent PR's being merged that do not follow the correct syntax.
+
+## Reverting a Release
+> If you have introduced a breaking bug in a release you have 2 options:
+> * If you have a fix immediately ready, commit and push it (or merge it via a pull request) to the release branch
+> * Otherwise, [revert the commit](https://git-scm.com/docs/git-revert) that introduced the bug and push the revert commit (or merge it via a pull request) to the release branch  
+> 
+> In both cases **semantic-release** will publish a new release, so your package users will get the fixed/reverted version.
+>
+> Depending on the package manager you are using, you might be able to un-publish or deprecate a release, in order to prevent users from downloading it by accident. For example, [NPM](https://www.npmjs.com/) allows you to [un-publish](https://docs.npmjs.com/cli/unpublish) [within 72 hours](https://www.npmjs.com/policies/unpublish) after release. You may also [deprecate](https://docs.npmjs.com/cli/deprecate) a release if you would rather avoid un-publishing.
```

Comment:
I'm not sure we want to suggest the option of unpublishing. If a user has immediately upgraded to a newly-published version of Carbon, then that version will be locked in their `package-lock.json` file, with the associated `resolved` URL (registry.npmjs.org) and `integrity` SHA512 hash of that version. If we subsequently unpublish that version, then I think `npm install` will crash with a `404 Not Found` error.

In my opinion, the option of unpublishing should only be used in extreme circumstances (such as accidentally publishing confidential data within the package, which must be permanently removed from public access).

Reply:
I've taken this paragraph verbatim from `semantic-release`. I'd like to leave this in here because it provides detail about removing git tags.